Test/Framework/Graphics/ModelTest.cs

```diff
@@ -74,6 +74,94 @@ public void CopyBoneTransformsTo_Exceptions()
             Assert.Throws<ArgumentNullException>(() => model.CopyBoneTransformsTo(null));
             Assert.Throws<ArgumentOutOfRangeException>(() => model.CopyBoneTransformsTo(new Matrix[0]));
         }
+
+        [Test]
+        public void ShouldDrawSampleModel()
+        {
+            var model = new SquareModel(gd);
+
+            PrepareFrameCapture();
+
+            model.Draw();
+
+            CheckFrames();
+        }
+
+        /// <summary>
+        /// Simple Model definition for a square. Created for testing purposes only.
+        /// 
+        /// The simple model is manually created with Model* related classes to cover them with unit tests.
+        /// </summary>
+        private sealed class SquareModel
+        {
+            // centre point of the model - used to point camera and calculate corners
+            private Vector3 centre = new Vector3(1, 5, 1);
+
+            // the model
+            private readonly Model model;
+
+            // matrixes required for visualisation.
+            private readonly Matrix view, projection;
+
+            /// <summary>
+            /// Creates a new instance of SquareModel.
+            /// </summary>
+            /// <param name="gd">GraphicsDevice uset to render visualisation.</param>
+            public SquareModel(GraphicsDevice gd)
+            {
+                view = Matrix.CreateLookAt(Vector3.Zero, centre, Vector3.Up);
+
+                projection = Matrix.CreatePerspectiveFieldOfView(MathHelper.PiOver4, gd.Viewport.AspectRatio, 0.1f, 100.0f);
+
+                // 4 vertices allow to construct a square.
+                // 1---2
+                // |  /|
+                // | c |
+                // |/  |
+                // 0---3
+                // where 'c' stands for 'centre'.
+                var vertices = new[]
+                {
+                    new VertexPosition(centre + new Vector3(-1,0,-1)),
+                    new VertexPosition(centre + new Vector3(-1,0,+1)),
+                    new VertexPosition(centre + new Vector3(+1,0,+1)),
+                    new VertexPosition(centre + new Vector3(+1,0,-1))
+                };
+                var vertexBuffer = new VertexBuffer(gd, VertexPosition.VertexDeclaration, vertices.Length, BufferUsage.None);
+                vertexBuffer.SetData(vertices);
+
+                var verticesPointers = new ushort[]
+                {
+                    0, 1, 2,
+                    0, 2, 3
+                };
+                var indexBuffer = new IndexBuffer(gd, IndexElementSize.SixteenBits, verticesPointers.Length, BufferUsage.None);
+                indexBuffer.SetData(verticesPointers);
+
+                var meshPart = new ModelMeshPart();
+                meshPart.NumVertices = vertices.Length;
```

Comment:
I suspect this test won't compile under XNA.  This is because at some point we made the model classes have setters where XNA does not.

What do you think of this @KonajuGames @Jjagg @dellis1972 ?